src/Features/Core/Portable/DisposeAnalysis/DisposableFieldsShouldBeDisposedDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,266 @@
+                {
+                    var fieldReference = (IFieldReferenceOperation)operationContext.Operation;
+                    var field = fieldReference.Field;
+
+                    // Check if this is a Disposable field that is not currently being tracked.
+                    if (_fieldDisposeValueMap.ContainsKey(field) ||
+                        !_disposableFields.Contains(field))
+                    {
+                        return;
+                    }
+
+                    // Only track instance fields on the current instance.
+                    if (field.IsStatic || fieldReference.Instance?.Kind != OperationKind.InstanceReference)
+                    {
+                        return;
+                    }
+
+                    // We have a field reference for a disposable field.
+                    // Check if it is being assigned a locally created disposable object.
+                    if (fieldReference.Parent is ISimpleAssignmentOperation simpleAssignmentOperation &&
+                        simpleAssignmentOperation.Target == fieldReference)
+                    {
+                        if (lazyPointsToAnalysisResult == null)
+                        {
+                            if (_disposeAnalysisHelper.TryGetOrComputeResult(operationBlockStartContext.OperationBlocks,
+                                containingMethod, operationBlockStartContext.Options, s_disposableFieldsShouldBeDisposedRule, trackInstanceFields: false,
+                                trackExceptionPaths: false, operationBlockStartContext.CancellationToken,
+                                out _, out var pointsToAnalysisResult) &&
+                                pointsToAnalysisResult != null)
+                            {
+                                Interlocked.CompareExchange(ref lazyPointsToAnalysisResult, pointsToAnalysisResult, null);
+                            }
+                        }
+
+                        PointsToAbstractValue assignedPointsToValue = lazyPointsToAnalysisResult[simpleAssignmentOperation.Value.Kind, simpleAssignmentOperation.Value.Syntax];
+                        foreach (var location in assignedPointsToValue.Locations)
+                        {
+                            if (_disposeAnalysisHelper.IsDisposableCreationOrDisposeOwnershipTransfer(location, containingMethod))
+                            {
+                                AddOrUpdateFieldDisposedValue(field, disposed: false);
+                                break;
+                            }
+                        }
+                    }
+                }
+
+                void AnalyzeDisposeMethod()
+                {
+                    // Perform dataflow analysis to compute dispose value of disposable fields at the end of dispose method.
+                    if (_disposeAnalysisHelper.TryGetOrComputeResult(operationBlockStartContext.OperationBlocks, containingMethod,
```

Comment:
so we also sharing whole dispose analysis itself on top of flow analysis engine?

Reply:
Yes, that is the core piece being shared.